savlatkhon-eshonov-1/task-4.ts

```diff
@@ -0,0 +1,186 @@
+interface Product {
+  name: string;
+  price: number;
+  stock: number
+}
+
+interface Customer {
+  name:string;
+  email:string;
+  address:string;
+}
+
+interface ShippingService {
+  shipOrder(order: Order): void;
+}
+
+class Product {
+  constructor(
```

Comment:
as we discussed in class, never expose class member as public. You are breaching security so these fields can be access from outside. as best practice, you should create getter and setter functions to update these members

`  getPrice() {
    return this.price;
  }

  getName() {
    return this.name;
  }

  getStock() {
    return this.stock;
  }`